src/cbrkit/sim/graphs/alignment.py

```diff
@@ -0,0 +1,191 @@
+        alignment = node_result.local_similarities
+
+        node_similarity = dist2sim(node_distance)
+
+        node_mappings = {
+            y_node.key: x_node.key for x_node, y_node in alignment if x_node and y_node
+        }
+
+        edge_similarity = None
+        edge_mappings = {}
+
+        if self.edge_sim_func:
+            edge_sequence_x = self.get_sequential_edges(x, sequence_x)
+            edge_sequence_y = self.get_sequential_edges(y, sequence_y)
+
+            # Use the dtwmodule for edge distances
+            edge_result = dtwmodule(distance_func=self.edge_sim_func)(
+                edge_sequence_x, edge_sequence_y, return_alignment=False
+            )
+
+            # Extract edge distance (value)
+            edge_distance = edge_result.value
+
+            edge_similarity = dist2sim(edge_distance)
+
+            edge_mappings = {
+                y_edge.key: x_edge.key
+                for x_edge, y_edge in zip(edge_sequence_x, edge_sequence_y)
+                if x_edge and y_edge
+            }
+
+        total_similarity = (
+            (node_similarity + edge_similarity) / 2.0
+            if edge_similarity is not None
+            else node_similarity
+        )
+
+        if self.normalize and len(alignment) > 0:
+            total_similarity /= len(alignment)
+
+        return GraphSim(
+            value=total_similarity,
+            node_mappings=node_mappings,
+            edge_mappings=edge_mappings,
+        )
+
+    def get_sequential_nodes(
+        self, graph: Graph[K, Any, Any, Any]
+    ) -> list[Node[K, Any]]:
+        """
+        Retrieves the nodes of the graph in sequential order.
+
+        Args:
+            graph: The graph to extract nodes from.
+
+        Returns:
+            A list of nodes in sequential order.
+        """
+        in_degree = {node.key: 0 for node in graph.nodes.values()}
+        for edge in graph.edges.values():
+            in_degree[edge.target.key] += 1
+        start_nodes = [
+            node for node in graph.nodes.values() if in_degree[node.key] == 0
+        ]
+        if len(start_nodes) != 1:
+            raise ValueError("Graph does not have a unique start node")
+        start_node = start_nodes[0]
+
+        sequence = []
+        current_node = start_node
+        visited_nodes = set()
+        while current_node and current_node.key not in visited_nodes:
+            sequence.append(current_node)
+            visited_nodes.add(current_node.key)
+            outgoing_edges = [
+                edge
+                for edge in graph.edges.values()
+                if edge.source.key == current_node.key
+            ]
+            if len(outgoing_edges) > 1:
+                raise ValueError(
+                    "Graph is not sequential (node has multiple outgoing edges)"
+                )
+            current_node = outgoing_edges[0].target if outgoing_edges else None
+        return sequence
+
+    def get_sequential_edges(
```

Comment:
It could make sense to combine the functions for sequential edges and nodes (because this function even requires the node sequence to be passed as an argument). Something like `to_sequence(graph: Graph) -> tuple[list[Node], list[Edge]]`